crates/driver/src/infra/notify/notification.rs

```diff
@@ -51,6 +51,8 @@ pub enum Kind {
         reason: BanReason,
         until: DateTime<Utc>,
     },
+    /// The solver sent an invalid request format
+    SerializationError(String),
```

Comment:
nit: technically the driver fails to deserialize the response. A `SerializationError` inside the driver would suggest it tried to send something to the solver or autopilot and was not able to serialize the request.
```suggestion
    DeserializationError(String),
```